emails/tests/models_tests.py

```diff
@@ -677,6 +695,78 @@ class ProfileHasPhoneTest(ProfileTestCase):
     def test_default_False(self) -> None:
         assert self.profile.has_phone is False
 
+    def test_phone_subscription_returns_True(self) -> None:
+        self.upgrade_to_phone_premium()
+        assert self.profile.has_phone is True
+
+
+@pytest.mark.skipif(not settings.PHONES_ENABLED, reason="PHONES_ENABLED is False")
+class ProfileDatePhoneRegisteredTest(ProfileTestCase):
+    """Tests for Profile.date_phone_registered"""
+
+    def setUp(self):
+        signals.post_save.disconnect(
+            sender=RealPhone, dispatch_uid="realphone_post_save"
+        )
+        return super().setUp()
+
+    def tearDown(self) -> None:
+        signals.post_save.connect(
+            receiver=realphone_post_save,
+            sender=RealPhone,
+            dispatch_uid="realphone_post_save",
+        )
+        return super().tearDown()
+
+    def test_default_None(self) -> None:
+        assert self.profile.date_phone_registered is None
+
+    def test_real_phone_no_relay_number_returns_verified_date(self) -> None:
+        self.upgrade_to_phone_premium()
+        datetime_now = datetime.now(timezone.utc)
+        RealPhone.objects.create(
+            user=self.profile.user,
+            number="+12223334444",
+            verified=True,
+            verified_date=datetime_now,
+        )
+        assert self.profile.date_phone_registered == datetime_now
+
+    @override_settings(PHONES_NO_CLIENT_CALLS_IN_TEST=True)
```

Comment:
Please move this up as a decorator for the class, and remove the `setUp` and `tearDown` methods

Reply:
Done 677eed23b0eaf1d2e599d63dab5b8bd12d83eeba